pkg/controller/config/config_controller.go

```diff
@@ -0,0 +1,344 @@
+package config
+
+import (
+	"context"
+	"flag"
+	"path/filepath"
+
+	"github.com/go-logr/logr"
+	mf "github.com/jcrossley3/manifestival"
+	op "github.com/openshift/tektoncd-pipeline-operator/pkg/apis/operator/v1alpha1"
+	"github.com/operator-framework/operator-sdk/pkg/predicate"
+	"github.com/prometheus/common/log"
+	appsv1 "k8s.io/api/apps/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	defaultResWatched = "cluster"
+	defaultTargetNs   = "openshift-pipelines-x" // TODO: -x
+)
+
+const (
+	installedStatus  = "installed"
+	installingStatus = "installing"
+	unknownStatus    = "unknown"
+	errorStatus      = "error"
+)
+
+var (
+	tektonVersion   = "v0.4.0"
+	resourceWatched string
+	resourceDir     string
+	targetNamespace string
+	noAutoInstall   bool
+	recursive       bool
+	ctrlLog         = logf.Log.WithName("config_controller")
+)
+
+func init() {
+	flag.StringVar(
+		&resourceWatched, "watch-resource", defaultResWatched,
+		"Name of the cluster-wide resource that operator honours, default: "+defaultResWatched)
+
+	flag.StringVar(
+		&targetNamespace, "target-namespace", defaultTargetNs,
+		"Namespace where pipeline will be installed default: "+defaultTargetNs)
+
+	defaultResDir := filepath.Join("deploy", "resources", tektonVersion)
+	flag.StringVar(
+		&resourceDir, "resource-dir", defaultResDir,
+		"Path to resource manifests, default: "+defaultResDir)
+
+	flag.BoolVar(
+		&noAutoInstall, "no-auto-install", false,
```

Comment:
will `skip-auto-install` or `disable-auto-install` be better

Reply:
`--no-` is a common convention